param/__init__.py

```diff
@@ -2978,7 +2978,25 @@ def _validate_order(self, val, step, allow_None):
                 f"start {end} with negative step {step}."
             )
 
-    def _validate_bounds(self, val, bounds, inclusive_bounds):
+    def _validate_bound_type(self, value, position, kind):
+        if not _is_number(value):
+            raise ValueError(
+                f"{_validate_error_prefix(self)} {position} {kind} can only be "
+                f"None or a numerical value, not {type(value)}."
+            )
+
+    def _validate_bounds(self, val, bounds, inclusive_bounds, softbounds):
+        if bounds is not None:
+            for pos, v in zip(['lower', 'upper'], bounds):
+                if v is None:
+                    continue
+                self._validate_bound_type(v, pos, 'bound')
+        if softbounds is not None:
+            for spos, sv in zip(['lower', 'upper'], softbounds):
+                if sv is None:
+                    continue
+                self._validate_bound_type(sv, spos, 'softbound')
+
```

Comment:
If you're changing the signature of this already, why not just change it to accept a bounds obj and a description, so that this code isn't duplicated? That way some future change to one of the code paths won't be missed on the other one.

Reply:
Refactored in https://github.com/holoviz/param/pull/824/commits/4a810acd2d41cd3835411214ad4c47b96a7b879b, I'll let you merge if you're okay with the change.